Lib/pathlib.py

```diff
@@ -1007,14 +1034,10 @@ def iterdir(self):
         The children are yielded in arbitrary order, and the
         special entries '.' and '..' are not included.
         """
-        for name in os.listdir(self):
-            yield self._make_child_relpath(name)
+        raise UnsupportedOperation(f"{type(self).__name__}.iterdir()")
 
     def _scandir(self):
-        # bpo-24132: a future version of pathlib will support subclassing of
-        # pathlib.Path to customize how the filesystem is accessed. This
-        # includes scandir(), which is used to implement glob().
-        return os.scandir(self)
+        return contextlib.nullcontext(list(self.iterdir()))
```

Comment:
This implementation of `_PathBase._scandir()` is new, and replaces a comment that alluded to its future addition. The `list()` call is necessary because `iterdir()` doesn't raise until you start iterating. The `iterdir()` behaviour was reported as a bug that we closed, but it could be revisited. See #78722.

Reply:
:+1: I'll do that if we decide against fixing #78722. My preference is still to adjust `iterdir()` rather than work around the problem in `_scandir()`

Reply:
I agree fixing iterdir at the source would be cleaner